### PR TITLE
54 get assignments

### PR DESCRIPTION
**Pull request by Aaya-Elsharief**

### CODE REVIEW

File: server/routes/class.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import { userAuth, studentAndTeacher } from '../middlewares';
-import { recommended, getAnnouncement } from '../controllers';
+import { recommended, getAnnouncement, studentAssignment } from '../controllers';
 
 const classRouter = require('express').Router();
 
 classRouter.get('/class/:classId/recommended', userAuth, studentAndTeacher, recommended);
 
 classRouter.get('/class/:classId/announcement', userAuth, studentAndTeacher, getAnnouncement);
+classRouter.get('/class/:classId/assignments/:studentId?', userAuth, studentAndTeacher, studentAssignment);
```
Comment (Raghad-Mezied) [inline server/routes/class.ts:9]: You will get the student id from the middleware, remove it

File: server/controllers/class/studentAssignment.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Response, NextFunction } from 'express';
+import { studentAssignmentQuery, teacherAssignmentQuery } from '../../queries';
+import { CustomError, CustomRequest } from '../../utils';
+
+const studentAssignment = async (req: CustomRequest, res: Response, next:NextFunction) => {
```
Comment (Raghad-Mezied) [inline server/controllers/class/studentAssignment.ts:5]: update the name, it's for both

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: server/middlewares/studentAndTeacher.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ const studentAndTeacher = (req: CustomRequest, res: Response, next: NextFunction
   const { user } = req;
 
   if (user.role === 'teacher' || user.role === 'student') {
+    req.user = user;
```
Comment (Raghad-Mezied) [inline server/middlewares/studentAndTeacher.ts:9]: You already passed the `userAuth` middleware where we add user data, so no need for this.

File: server/queries/class/teacherAssignmentQuery.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+const teacherAssignmentQuery = async (classId : string) => AssignmentStudent
+  .findAndCountAll({
+
+    raw: true,
+    nest: false,
+    attributes: ['Assignment.id' as 'id', 'Assignment.title' as 'title',
+    'Assignment.description' as 'description', 'Assignment.class_id' as 'class_id'],
+    where: {
+    },
```
Comment (Raghad-Mezied) [inline server/queries/class/teacherAssignmentQuery.ts:11]: If there is no condition then don't write it

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Bas-Shiekh — DISMISSED** [review]

**Review by Raghad-Mezied — DISMISSED** [review]

**Review by moustf — APPROVED** [review]

**Pull request merged**